app/models/manageiq/providers/azure/inventory/collector/target_collection.rb

```diff
@@ -521,7 +521,7 @@ def infer_related_vm_ems_refs_db!
   def infer_related_vm_ems_refs_api!
     instances.each do |instance|
       # TODO(lsmola) add API scanning
-      target.add_target(:association => :flavors, :manager_ref => {:name => instance.properties.hardware_profile.vm_size.downcase})
+      target.add_target(:association => :flavors, :manager_ref => {:name => instance.properties.hardware_profile.vm_size})
```

Comment:
I tried switching this to use `ems_ref` with the downcase'd version, but that seemed to make it unhappy.

Reply:
May that be caused by using `:name` as `:manager_ref`? Shouldn't you rather use `:ems_ref`? I mean like this:

```ruby
target.add_target(:association => :flavors, :manager_ref => {:ems_ref => instance.properties.hardware_profile.vm_size.downcase})
```

Reply:
yeah, looks like flavor has defined ```:manager_ref => [:ems_ref]``` so we should use ems_ref ( this looks like a bug )

Reply:
parser code for graph refresh is here, it worked because ems_ref and name are the same https://github.com/Ladas/manageiq-providers-azure/blob/949a23dd25ab883a87fe5d5b430442d135e9434d/app/models/manageiq/providers/azure/inventory/parser/cloud_manager.rb#L36

Reply:
@tumido @Ladas I've updated that to point at the ems_ref instead.